### PR TITLE
Add spec for nil?

### PR DESCRIPTION
**Pull request by sampatbadhe**
Add spec for nil?

### CODE REVIEW

File: core/kernel/nil_spec.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
 require_relative '../../spec_helper'
 require_relative 'fixtures/classes'
 
-describe "Kernel#nil?" do
-  it "needs to be reviewed for spec completeness"
+describe 'Kernel#nil?' do
+  it 'returns false for other than nil' do
```
Comment (eregon) [inline core/kernel/nil_spec.rb:5]: Actually it's simply `returns false`.
There are `NilClass#nil?` specs in https://github.com/ruby/spec/blob/master/core/nil/nil_spec.rb

Comment (sampatbadhe) [reply core/kernel/nil_spec.rb:5]: Thank you for reviewing @eregon 

> Actually it's simply returns false.

Sure, I'll update.

> There are NilClass#nil? specs in https://github.com/ruby/spec/blob/master/core/nil/nil_spec.rb

Thanks for the reference, I'll remove the below spec.

Comment (sampatbadhe) [reply core/kernel/nil_spec.rb]: Made the requested changes. @eregon could you please review?

**Review by eregon — APPROVED** [review]
Thanks!

**Pull request merged**